src/dialogs/voice-assistant-setup/voice-assistant-setup-step-local.ts

```diff
@@ -0,0 +1,345 @@
+import { listSTTEngines } from "../../data/stt";
+import { listTTSEngines, listTTSVoices } from "../../data/tts";
+import type { HomeAssistant } from "../../types";
+import { documentationUrl } from "../../util/documentation-url";
+import { AssistantSetupStyles } from "./styles";
+import { STEP } from "./voice-assistant-setup-dialog";
+
+@customElement("ha-voice-assistant-setup-step-local")
+export class HaVoiceAssistantSetupStepLocal extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ attribute: false })
+  public assistConfiguration?: AssistSatelliteConfiguration;
+
+  @state() private _state: "INSTALLING" | "NOT_SUPPORTED" | "ERROR" | "INTRO" =
+    "INTRO";
+
+  @state() private _detailState?: string;
+
+  @state() private _localTts?: EntityRegistryDisplayEntry[];
+
+  @state() private _localStt?: EntityRegistryDisplayEntry[];
+
+  protected override render() {
+    return html`<div class="content">
+      ${this._state === "INSTALLING"
+        ? html`<img src="/static/images/voice-assistant/update.png" />
+            <h1>Installing add-ons</h1>
+            <p>
+              The Whisper and Piper add-ons are being installed and configured.
+            </p>
+            <ha-circular-progress indeterminate></ha-circular-progress>
+            <p>
+              ${this._detailState || "Installation can take several minutes"}
+            </p>`
+        : this._state === "ERROR"
+          ? html` <img src="/static/images/voice-assistant/error.png" />
+              <h1>Failed to install add-ons</h1>
+              <p>
+                We could not automatically install a local TTS and STT provider
+                for you. Read the documentation to learn how to install them.
+              </p>
+
+              <a
+                href=${documentationUrl(
+                  this.hass,
+                  "/voice_control/voice_remote_local_assistant/"
+                )}
+                target="_blank"
+                rel="noreferrer noopenner"
+              >
+                <ha-button @click=${this._skip}>
+                  <ha-svg-icon .path=${mdiOpenInNew} slot="icon"></ha-svg-icon>
+                  Learn more</ha-button
+                >
+              </a>`
+          : this._state === "NOT_SUPPORTED"
+            ? html`<img src="/static/images/voice-assistant/error.png" />
+                <h1>Installation of add-ons is not supported on your system</h1>
+                <p>
+                  Your system is not supported to automatically install a local
+                  TTS and STT provider. Learn how to set up local TTS and STT
+                  providers in the documentation.
+                </p>
+                <a
+                  href=${documentationUrl(
+                    this.hass,
+                    "/voice_control/voice_remote_local_assistant/"
+                  )}
+                  target="_blank"
+                  rel="noreferrer noopenner"
```

Comment:
```suggestion
                  rel="noreferrer noopener"
```

I think there are 2 more in the code, maybe fix all.